packages/@angular/cli/commands/build.ts

```diff
@@ -25,6 +25,12 @@ export const baseBuildCommandOptions: any = [
   { name: 'locale', type: String },
   { name: 'extract-css', type: Boolean, aliases: ['ec'] },
   {
+    name: 'compress',
+    type: Boolean,
+    default: true,
```

Comment:
Except the default is false for the development target which makes it misleading in its current form.

Help should eventually support displaying target specific defaults.